### DISCUSSION TITLE
Windows App SDK Terminology - How to Understand?

### DISCUSSION THREAD

**Discussion by maptz** [microsoft/WindowsAppSDK#1791] [Q&A] [answered]
Hi, 

I am right royally confused about the terminology around Windows App SDK and WinUI 3. 

If I look at the [documentation](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app?tabs=csharp%2Ccsharp-unpackaged) , I see I can create a new "WinUI 3 in Desktop" app.  If I do that, the resulting file has got some XAML in with a project structure that looks a bit like UWP or WPF xaml. 

Is this a WinUI 3 app or a Windows App SDK app? 

If I run into a problem and want to search the internet for a solution, what is the right terminology here? For instance, I want to look up the `Binding` syntax and understand how it differs from WPF binding syntax. What would my Google search be?

**Comment by BenJKuhn** [comment, marked as answer]
Let me see if I can help.

The Windows App SDK is a set of application development tools and associated runtime components to help developers create great applications that run on Windows. WinUI 3 is the latest generation XAML-based UI framework, and is made available as part of the Windows App SDK. Developers often characterize their app based on the UI framework chosen, since it's such a defining characteristic. So they might say they have a WPF app or a WinForms app. In that sense, an app that uses WinUI 3 via Windows App SDK is a WinUI3 app.

That same app would have build and runtime dependencies on the Windows App SDK. So in that sense, it is also fair to say it's a Windows App SDK app. That's true whether it's only using WinUI 3 or using a wide range of Windows App SDK features. Similarly, a console application could take dependencies on the Windows App SDK to use, say, MRTCore support for localization but not have any dependencies on WinUI 3. That app could still be said to be a Windows App SDK app, even though it's not a WinUI 3 app.

This is a great question, and an easy one to get confused about. Thanks for asking!

Ben

**Reply by andrewleader** [reply]
Additionally, @maptz, to answer your question about what to search for, if you have a `Binding` syntax question, you should search for `WinUI 3 binding`, as the binding syntax is a WinUI 3 feature.

If you have a question about the AppWindow API, you would search for `AppWindow Windows App SDK`.

**Comment by maptz** [comment]
Thanks both of you for the excellent answer. I understand now. 

As an observation, it's very tricky to onboard here. I'm a hobbyist developer, but I'm experienced with WPF, WinForms and UWP. I've been slow to adopt any of the new GUI stacks mainly because the terminology is so difficult to parse. That's before I've even touched any code. 

Perhaps there's something for the team to look at? 

Would a very simple website which outlines the relationship between the following technologies help:

- MAUI, UWP, WinRT, WinUI, Windows App SDK, Xamarin Forms?

There's something amazing happening with all these options for building (cross-platform) GUIs coming from the .net team, but the complexity is amazingly difficult if you're looking at it from the outside. 

Without a simple and clear pathway to entry, adoption will be damagingly slow.
